### DISCUSSION TITLE
VSCode Code Language Not supported or Defined

### DISCUSSION THREAD

**Discussion by parsonsdjp** [quarto-dev/quarto-cli#5385] [Q&A]
Sorry if this is a basic question; I'm just getting started.
I use VSCode for most of my programming and haven't come across this problem until now.

I have downloaded quarto and installed the VSCode extension for it. When I create a new .qmd file, I receive the error "Code language not supported or defined" in the bottom right hand corner. However, I can see that "Quarto" is indeed my selected language (per the screenshot) and that the file extension of .qmd is valid.

I do not experience this problem if I run the same files in, say, RStudio. The problem seems to be how Quarto is interacting with the coderunner extension, but I'm a little lost in determining what the problem is.

Can anyone advise what steps I should take?

![image](https://user-images.githubusercontent.com/101063922/235683676-92d64ae1-1ae2-47d1-a546-d1f351fb114b.png)

**Comment by mcanouil** [comment]
Note that your error here has nothing to do with Quarto (CLI or VSCode extension), it's an error from "Code Runner" extension (as stated in the popup) which does not support Quarto.
Either disable this extension for Quarto documents or reach out to the authors of this extension to ask them if they can support Quarto (as a derivative of Markdown).
